assets/js/modules/analytics/components/settings/SettingsEdit.js

```diff
@@ -46,6 +49,61 @@ export default function SettingsEdit() {
 		select( CORE_SITE ).isUsingProxy()
 	);
 
+	const loggedInUserID = useSelect( ( select ) =>
+		select( CORE_USER ).getID()
+	);
+	const hasResolvedUser = useSelect( ( select ) =>
+		select( CORE_USER ).hasFinishedResolution( 'getUser' )
+	);
+
+	const hasAnalyticsAccess = useSelect( ( select ) => {
+		const moduleOwnerID = select( MODULES_ANALYTICS ).getOwnerID();
+
+		if ( moduleOwnerID === loggedInUserID ) {
+			return true;
+		}
+		return select( CORE_MODULES ).hasModuleAccess( 'analytics' );
+	} );
+	const hasResolvedAnalyticsAccess = useSelect( ( select ) => {
+		const hasResolvedModuleOwner = select(
+			MODULES_ANALYTICS
+		).hasFinishedResolution( 'getSettings' );
+
+		const isResolvingModuleAccess = select(
+			CORE_MODULES
+		).isResolving( 'hasModuleAccess', [ 'analytics' ] );
```

Comment:
Your variables here are indicating [has resolved] but the selector you're using here is only whether or not it is currently resolving. Looks like maybe this was intentional but I think it should also use `hasFinishedResolution`.

Reply:
`hasFinishedResolution` is false even when resolution hasn't started - hence I had to use `isResolving` which is true only if resolution started.

Reply:
I have renamed the variables and updated the logic to make the check more readable.